packages/shared/src/lib/core/token/utils/convertToRawAmount.ts

```diff
@@ -18,29 +16,37 @@ function convertToRawAmountFromMetadata(
     amount: number,
     tokenMetadata: TokenMetadata,
     selectedUnit?: string
-): Big | undefined {
+): bigint | undefined {
+    const [min, max] = bigIntMinAndMax(tokenMetadata.decimals, MAX_SUPPORTED_DECIMALS)
+
     if (tokenMetadata?.standard === TokenStandard.BaseToken) {
-        if (tokenMetadata.useMetricPrefix) {
-            const decimals = IOTA_UNIT_MAP?.[selectedUnit?.substring(0, 1) as IotaUnit]?.decimalPlaces ?? 0
-            return convertAmountToMatchUnit(amount, decimals)
+        if (!selectedUnit || selectedUnit === tokenMetadata.unit) {
+            return convertFloatToBigInt(amount, max)
```

Comment:
i needed to change this to `min` so that it works correctly, else it converted 1SMR into 1 * 10^24 GLOW. Any reason for this?